drake/systems/visualizers/Visualizer.m

```diff
@@ -127,13 +127,20 @@ function playback(obj,xtraj,options)
 
       % set up continuous slider feedback:
       time_slider_listener = addlistener(time_slider,'ContinuousValueChange',@update_time_display);
-
+      playback_start_time = 0;
       function update_speed(source, eventdata)
-        obj.playback_speed = 10 ^ (get(speed_slider, 'Value'));
+        new_playback_speed = 10 ^ get(speed_slider, 'Value');
+
+        % adjust the start time to prevent simulation time changing
+        currtime = cputime();
+        playback_start_time = currtime - (obj.playback_speed / new_playback_speed) *(currtime - playback_start_time);
+
+        obj.playback_speed = new_playback_speed;
         set(speed_display, 'String', sprintf(speed_format, obj.playback_speed));
       end
       function rewind_vis(source, eventdata)
         set(time_slider, 'Value', get(time_slider, 'Min'));
+        playback_start_time = cputime();
```

Comment:
I think what @RussTedrake is suggesting is

```matlab
a = tic;
playback_start_time = toc(a);
...
t=t0+obj.playback_speed*(toc (a) - playback_start_time);
```
You can call `toc` multiple times for one timerval. `cputime` does not measure wall clock time, which is what you want here.